lib/contracts/builtin_contracts.rb

```diff
@@ -405,6 +405,7 @@ def initialize(contract_hash)
 
       def valid?(arg)
         return false unless arg.is_a?(Hash)
+        return false unless arg.keys.sort.eql?(contract_hash.keys.sort)
```

Comment:
I think we can drop support for ruby 1.8.7, I like comparing the keys instead of array size though.